api/lanes.py

```diff
@@ -562,3 +590,59 @@ def apply_filters(self, qu, work_model=Work, *args, **kwargs):
         qu = qu.join(work_edition).filter(work_edition.series==self.series)
         qu = qu.order_by(work_edition.series_position, work_edition.title)
         return qu
+
+
+class ContributorLane(QueryGeneratedLane):
+    """A lane of Works written by a particular contributor"""
+
+    ROUTE = 'contributor'
+    MAX_CACHE_AGE = 48*60*60    # 48 hours
+
+    def __init__(self, _db, contributor_name, contributor_id=None):
+        if not contributor_name:
+            raise ValueError("ContributorLane can't be created without contributor")
+
+        self.contributor_name = contributor_name
+        self.contributor = None
+        if contributor_id:
+            self.contributor = get_one(_db, Contributor, id=contributor_id)
+            if (self.contributor_name!=self.contributor.display_name and
+                self.contributor_name!=self.contributor.name):
+                raise ValueError(
+                    "ContributorLane can't be created with inaccurate"
+                    " Contributor data."
+                )
+
+        full_name = display_name = "Books by %s" % self.contributor_name
+        super(ContributorLane, self).__init__(
+            _db, full_name, display_name=display_name
+        )
+
+    @property
+    def url_arguments(self):
+        kwargs = dict(contributor_name=self.contributor_name)
+        return self.ROUTE, kwargs
+
+    def apply_filters(self, qu, work_model=Work, *args, **kwargs):
+        if not self.contributor_name:
+            return None
+        qu = self.only_show_ready_deliverable_works(qu, work_model)
+
+        work_edition = aliased(Edition)
+        qu = qu.join(work_edition).join(work_edition.contributions)
+        qu = qu.join(Contribution.contributor).order_by(False)
```

Comment:
This was necessary when I was experimenting with unions, but no longer. *&#42;deletes aggressively&#42;*